TACC/Core-Portal
# task/FP-585 - Show message when allocation missing for storage system

## Overview: ##

Display an error when a user is missing an allocation for a requested storage system

## PR Status: ##


## Related Jira tickets: ##

* [FP-585](https://jira.tacc.utexas.edu/browse/FP-585)

## Summary of Changes: ##

- datafiles get view checks error response status code for 502 (unable to get file listing)
- if response is 502 and the user is missing the tas allocation for the associated host, return HTTPError 403 (which is displayed in the react client with a "requires allocation" message)
- if response is 502 and the user has the required tas allocation, return the response that will cause system push keys modal (this is existing code from a different PR)
- unit tests for both cases

## Testing Steps: ##
1. Use one of the test accounts that is missing an allocation for Frontera and browse to Data Files. You should see a message saying that you are missing the required allocation
2. Use a test account that DOES have the required allocation and make sure a push keys modal shows up.

## UI Photos:

![image](https://user-images.githubusercontent.com/17181582/90556696-f31a8b80-e15e-11ea-937e-e8e24ca84641.png)
